### ISSUE TITLE
[vsdebug] Stop execution on breakpoints across all threads when running in CLI debug mode

### ISSUE THREAD

**Issue by PranayAgarwal** [facebook/hhvm#8218]
### HHVM Version

3.25.3/3.26.3

### Operating System and Version

macOS High Sierra/Ubuntu 16.04/Ubuntu 18.04

### Problem 

When hhvm is started in `--mode vsdebug`, it currently only stops on breakpoints in the main execution thread. If the same code path is hit from an alternate thread (e.g. when run using `xbox_task_start()`), the breakpoint is ignored.

Execution *can* be halted from other threads, however, by putting `hphp_debug_break()` statements (but stepping into/over from there doesn't work).

### How to repro

**`main.php`:**

```php
<?hh

function run() {
    $i = 1;
    while (true){
        $resource = xbox_task_start((string)$i++);

        $output = null;
        $result = xbox_task_result($resource, 0, &$output);
        echo "Execution result: $result\n";
        echo "Task output: $output\n";
    }
}

run();
```

**`worker.php`:**

```php
<?hh

function xbox_process_message(string $task): string {
    echo "Running task $task\n";
    // hphp_debug_break();
    sleep(1);
    return "Task $task finished";
}
```

Start this using: `hhvm -d hhvm.xbox.server_info.request_init_document=worker.php --mode vsdebug --vsDebugPort 8999 --vsDebugNoWait true main.php`

Attach to the process in vscode using the default hhvm attach config. You should now see the tasks continue executing in the vscode debug console.

Set a breakpoint anywhere in the while loop in main.php. That breakpoint will immediately be hit. You can also see the new thread(s) spawned for the task in the call stack panel.

Set a breakpoint in `xbox_process_message` in worker.php. That breakpoint will never be hit.

Uncomment the `// hphp_debug_break();` line in worker.php. Execution will now halt in the job runner on that line, but you won't be able to step through.

**Comment by fredemmott** [comment]
cc @ebluestein - how feasible is this? I was assuming that this would basically be cross-request-debugging, but hphp_debug_break() working makes it sound like that's not the case?

**Comment by ebluestein** [comment]
Thanks for the heads up, this is not a scenario I have tried previously. Since hphpd_debug_break does work, we are able to stop the request in the debugger. But since stepping doesn't work, my guess is the debugger is not tracking this type of request properly. It probably differs in how it starts up from how a regular request starts and I'd guess I'm missing it.

I'll have to investigate this one a little but I think it should be possible to make this work

**Comment by ebluestein** [comment]
This should be fixed by 56b2c4c4584378427311493340f83919daaf3538!
